pandas/core/indexes/multi.py

```diff
@@ -3048,22 +3055,25 @@ def convert_indexer(start, stop, step, indexer=indexer, codes=level_codes):
 
         else:
 
-            code = self._get_loc_single_level_index(level_index, key)
+            idx = self._get_loc_single_level_index(level_index, key)
```

Comment:
side-note: _get_loc_single_level_index looks sketchy.  if level_index contains NA values, we should probably be doing get_loc and not returning -1.

Reply:
Will look into this as a follow up

Reply:
Have looked into this. Problem is, we are taking an index level from the MultiIndex, which does not contain the nans, so we will always get a KeyError, even if the original MultiIndex had nan in this level.